--- buildroot/package/freerdp/freerdp.hash.orig
+++ buildroot/package/freerdp/freerdp.hash
@@ -1,5 +1,5 @@
-# From https://pub.freerdp.com/releases/freerdp-2.3.1.tar.gz.sha256
-sha256  20071c691c1947ca256149e1d03edec1a5dfbf9960f0ef4e21ecab87433c2178  freerdp-2.3.1.tar.gz
+# From https://pub.freerdp.com/releases/freerdp-2.3.2.tar.gz.sha256
+sha256  deb888034a441c7f76dc8b3ddea67fac3c0d815739fc2146e1243480ce56c91c  freerdp-2.3.2.tar.gz
 
 # Locally calculated
 sha256  cfc7749b96f63bd31c3c42b5c471bf756814053e847c10f3eb003417bc523d30  LICENSE
--- buildroot/package/freerdp/freerdp.mk.orig
+++ buildroot/package/freerdp/freerdp.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-FREERDP_VERSION = 2.3.1
+FREERDP_VERSION = 2.3.2
 FREERDP_SITE = https://pub.freerdp.com/releases
 FREERDP_DEPENDENCIES = libglib2 openssl zlib
 FREERDP_LICENSE = Apache-2.0
