--- buildroot/package/python-glslang/python-glslang.hash.orig
+++ buildroot/package/python-glslang/python-glslang.hash
@@ -1,3 +1,3 @@
 # Locally calculated
-sha256  c6c21fe1873c37e639a6a9ac72d857ab63a5be6893a589f34e09a6c757174201  python-glslang-15.3.0.tar.gz
+sha256  592c98aeb03b3e81597ddaf83633c4e63068d14b18a766fd11033bad73127162  python-glslang-11.13.0.tar.gz
 sha256  adb783e734e906d1f46db5df29991dbde84bdb0ceab502ac2febb44fe3c2b5f4  LICENSE.txt
--- buildroot/package/python-glslang/python-glslang.mk.orig
+++ buildroot/package/python-glslang/python-glslang.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-PYTHON_GLSLANG_VERSION = 15.3.0
+PYTHON_GLSLANG_VERSION = 11.13.0
 PYTHON_GLSLANG_SITE = $(call github,KhronosGroup,glslang,$(PYTHON_GLSLANG_VERSION))
 PYTHON_GLSLANG_LICENSE = BSD-3-Clause
 PYTHON_GLSLANG_LICENSE_FILES = LICENSE.txt
